pkg/discovery/dtofactory/pod_entity_dto_builder.go

```diff
@@ -290,6 +299,30 @@ func (builder *podEntityDTOBuilder) getPodCommoditiesBoughtFromQuota(quotaUID st
 	return commoditiesBought, nil
 }
 
+// Build the CommodityDTOs bought by the pod from the Volumes.
+func (builder *podEntityDTOBuilder) getPodCommoditiesBoughtFromVolumes(pod *api.Pod, mounts []mountedVolumes) ([]*proto.CommodityDTO, error) {
+	var commoditiesBought []*proto.CommodityDTO
+	podKey := util.PodKeyFunc(pod)
+
+	for _, mount := range mounts {
+		mountName := mount.mountName
+		volEntityID := util.PodVolumeMetricId(podKey, mountName)
+		// We use <ns>/<pod-name>/<mounted-vol-name> as the commodity key
+		// to keep the relationship between pod and volume unique.
+		commKey := fmt.Sprintf("%s/%s", podKey, mountName)
```

Comment:
StorageAmount is usually not key-ed. What's the purpose of having this unique key here?

Reply:
As in the comment above. The intention is to keep the buy sell relationship between a pod and a volume unique and so actions are not generated for pods across volumes provising the same commodity (STORAGE_AMOUNT). However now that you already mention it; there are more ways of doing this (set movable to false on entities), and more scenarios that I need ways to handle (and am sort of struggling with).

For starters the mapping between pods and volumes seems to be many to many:
- The same volume can be mounted by multiple pods, depending on the readwrite modes and the support from underlying volume provider.
- One pod can mount multiple volumes.
- If the volume is mounted by multiple pods, the usage value is not supposed to be aggregated, rather each pod mounting the same volume would report the same usage amount.

I will reach out to you, if I can't arrive at a clean way of achieving the above.